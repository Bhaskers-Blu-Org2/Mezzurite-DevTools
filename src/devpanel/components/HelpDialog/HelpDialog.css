@media only screen {
  .help-dialog--action {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
  }

  .help-dialog--cancel {
    background: none;
    border: none;
    font-size: 20px;
    margin: 8px;
    position: absolute;
    right: 0;
    top: 0;
  }

  .help-dialog--close {
    background-color: var(--color-blue);
    border: none;
    border-radius: 2px;
    color: var(--color-white);
    font-size: 14px;
    padding: 12px 36px;
  }

  .help-dialog--close:active,
  .help-dialog--close:hover {
    background-color: var(--color-blue-darker);
    cursor: pointer;
    text-transform: underline;
  }

  .help-dialog--container {
    align-content: center;
    display: flex;
    justify-content: center;
    min-width: 320px;
    padding: 16px;
    width: 100%;
  }

  .help-dialog--content {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0 28px;
    width: 100%;
  }

  .help-dialog--heading {
    font-weight: 600;
    text-align: center;
  }

  .help-dialog--item {
    line-height: 18px;
    margin-bottom: 20px;
    font-size: 14px;
  }

  .help-dialog--learn-more {
    color: var(--color-blue);
  }

  .help-dialog--vocab-title {
    font-weight: 600;
  }
}

@media only screen and (min-width: 640px) {
  .help-dialog--action {
    align-items: flex-end;
  }
}
