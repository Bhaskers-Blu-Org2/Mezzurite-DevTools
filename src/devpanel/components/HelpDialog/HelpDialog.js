import React from 'react';
import { func } from 'prop-types';

import './HelpDialog.css';

const applicationLoadTimeDescription = ` is the amount of time it takes to load the SPA framework. This is calculated using the navigation start time and the application framework load time.`;
const componentLoadTimeDescription = ` is the amount of time it takes to load a single component. This is calculated by using the framework's component life cycle.`;
const viewportLoadTimeDescription = `  is the amount of time it takes to load the components visible in the current viewport.`;

const HelpDialog = (props) => (
  <div className='help-dialog--container'>
    <div className='help-dialog--content'>
      <h2 className='help-dialog--heading'>Understanding Mezzurite Metrics</h2>
      <ul className='help-dialog--vocab'>
        <li className='help-dialog--vocab-item'>
          <span className='help-dialog--vocab-title'>Application Load Time (ALT)</span>
          {applicationLoadTimeDescription}
        </li>
        <li className='help-dialog--vocab-item'>
          <span className='help-dialog--vocab-title'>Component Load Time (CLT)</span>
          {componentLoadTimeDescription}
        </li>
        <li className='help-dialog--vocab-item'>
          <span className='help-dialog--vocab-title'>Viewport Load Time (VLT)</span>
          {viewportLoadTimeDescription}
        </li>
      </ul>
      <div className='help-dialog--action'>
        <a
          aria-label='Learn more about Mezzurite'
          className='help-dialog--learn-more'
          href='https://github.com/Microsoft/Mezzurite#background'
          target='_blank'
        >
          Learn More
        </a>
        <button
          aria-label='Close'
          className='help-dialog--close'
          onClick={props.onCloseClick}
        >
          Close
        </button>
      </div>
    </div>
  </div>
);

HelpDialog.propTypes = {
  onCloseClick: func
};

export default HelpDialog;
