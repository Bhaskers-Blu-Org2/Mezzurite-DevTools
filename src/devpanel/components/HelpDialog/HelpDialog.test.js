import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HelpDialog from './HelpDialog';

describe('HelpDialog.js', () => {
  const renderer = new ShallowRenderer();

  it('should render without an onCloseClick function', () => {
    const tree = renderer
      .render(<HelpDialog />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with an onCloseClick function', () => {
    const tree = renderer
      .render(<HelpDialog onCloseClick={() => true} />);
    expect(tree).toMatchSnapshot();
  });
});
