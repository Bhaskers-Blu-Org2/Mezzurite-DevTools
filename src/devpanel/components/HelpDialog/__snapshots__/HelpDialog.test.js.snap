// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDialog.js should render with an onCloseClick function 1`] = `
<React.Fragment>
  <button
    className="help-dialog--cancel"
    onClick={[Function]}
  >
    <i
      aria-label="Close"
      className="ms-Icon ms-Icon--Cancel"
      title="Close"
    />
  </button>
  <div
    className="help-dialog--container"
  >
    <div
      className="help-dialog--content"
    >
      <h2
        className="help-dialog--heading"
      >
        Understanding Mezzurite Metrics
      </h2>
      <ul
        className="help-dialog--vocab"
      >
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Application Load Time (ALT)
          </span>
           is the amount of time it takes to load the SPA framework. This is calculated using the navigation start time and the application framework load time.
        </li>
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Component Load Time (CLT)
          </span>
           is the amount of time it takes to load a single component. This is calculated by using the framework's component life cycle.
        </li>
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Viewport Load Time (VLT)
          </span>
            is the amount of time it takes to load the components visible in the current viewport.
        </li>
        <li
          className="help-dialog--item"
        >
          <a
            aria-label="Learn more about Mezzurite"
            className="help-dialog--learn-more"
            href="https://github.com/Microsoft/Mezzurite#background"
            target="_blank"
          >
            Learn More
          </a>
        </li>
      </ul>
      <div
        className="help-dialog--action"
      >
        <button
          aria-label="Close"
          className="help-dialog--close"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>
</React.Fragment>
`;

exports[`HelpDialog.js should render without an onCloseClick function 1`] = `
<React.Fragment>
  <button
    className="help-dialog--cancel"
  >
    <i
      aria-label="Close"
      className="ms-Icon ms-Icon--Cancel"
      title="Close"
    />
  </button>
  <div
    className="help-dialog--container"
  >
    <div
      className="help-dialog--content"
    >
      <h2
        className="help-dialog--heading"
      >
        Understanding Mezzurite Metrics
      </h2>
      <ul
        className="help-dialog--vocab"
      >
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Application Load Time (ALT)
          </span>
           is the amount of time it takes to load the SPA framework. This is calculated using the navigation start time and the application framework load time.
        </li>
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Component Load Time (CLT)
          </span>
           is the amount of time it takes to load a single component. This is calculated by using the framework's component life cycle.
        </li>
        <li
          className="help-dialog--item"
        >
          <span
            className="help-dialog--vocab-title"
          >
            Viewport Load Time (VLT)
          </span>
            is the amount of time it takes to load the components visible in the current viewport.
        </li>
        <li
          className="help-dialog--item"
        >
          <a
            aria-label="Learn more about Mezzurite"
            className="help-dialog--learn-more"
            href="https://github.com/Microsoft/Mezzurite#background"
            target="_blank"
          >
            Learn More
          </a>
        </li>
      </ul>
      <div
        className="help-dialog--action"
      >
        <button
          aria-label="Close"
          className="help-dialog--close"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</React.Fragment>
`;
